core/runtime.rs

```diff
@@ -2399,17 +2428,17 @@ pub fn queue_async_op<'s>(
     Some(scope.get_current_context())
   );
 
-  match OpCall::eager(op) {
+  match OpCall::eager(ctx.realm_idx, promise_id, ctx.id, op) {
     // If the result is ready we'll just return it straight to the caller, so
     // we don't have to invoke a JS callback to respond. // This works under the
     // assumption that `()` return value is serialized as `null`.
-    EagerPollResult::Ready((_, _, op_id, mut resp)) if !deferred => {
-      let resp = resp.to_v8(scope).unwrap();
-      ctx.state.borrow_mut().tracker.track_async_completed(op_id);
+    EagerPollResult::Ready(mut op_result) if !deferred => {
```

Comment:
I'm not sure this makes sense... If the op is deferred, why are we eagerly polling it? Seems like it would be better to push it directly to `pending_ops` container and let it be polled on the next tick?

Reply:
If we did that we can probably remove one level of indirection with `OpCall::ready`?